etcd/src/main/scala/io/buoyant/etcd/Key.scala

```diff
@@ -0,0 +1,210 @@
+package io.buoyant.etcd
+
+import com.twitter.finagle.{Path, Service}
+import com.twitter.finagle.http._
+import com.twitter.io.Buf
+import com.twitter.util._
+import scala.collection.JavaConverters._
+
+class Key(key: Path, client: Service[Request, Response]) {
+
+  import Etcd._
+
+  def path: Path = key
```

Comment:
Using Finagle Paths as urls seems like a weird coupling.  Why not just `Seq[String]`?